chia/full_node/lock_queue.py

```diff
@@ -54,18 +46,26 @@ class LockQueue(Generic[_T_Comparable]):
     Must be created while an asyncio loop is running.
     """
 
-    _queue: queue.PriorityQueue[_Element[_T_Comparable]] = dataclasses.field(default_factory=queue.PriorityQueue)
-    _active: Optional[_Element[_T_Comparable]] = None
-    _creation_counter: int = 0
-    cancelled: Set[_Element[_T_Comparable]] = dataclasses.field(default_factory=set)
+    _deques: Dict[int, collections.deque[_Element]]
+    _priority_index_map: Dict[_T_Priority, int]
```

Comment:
it seems like you could combine these two dictionaries into one:
```
_deques: Dict[_T_Priority, collections.deque[_Element]]
```
I imagine you started off with `_deques` being a `List`, which would require you to map enums to dense indices.